### PR TITLE
Fixed and Updated Documentation References in Examples/Core Sunpy Code 

### PR DESCRIPTION
**Pull request by SuleimanFarah**
Changed applicable coding documentation references to include valid DOIs from the original url form (primarily in the sunpy/sunpy code and sunpy/example files). Formatted the code for possible word choice errors or spelling mistakes. Researched new up-to-date links by the same author for references that led to expired websites.

This pull request was completed in an attempt to work on the #3273 issue. Thank you for any feedback on this pull request for many more to come!

### CODE REVIEW

File: changelog/6903.doc.rst [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Changed applicable coding documentation references to include valid DOIs from the original url form
+(primarily in the sunpy/sunpy code and sunpy/example files).
+Formatted the code for possible word choice errors or spelling mistakes.
+Researched new up-to-date links by the same author for references that
+led to expired websites.
+
+This pull request was completed in an attempt to work on the #3273 issue.
```
Comment (nabobalis) [inline changelog/6903.doc.rst:7]:
```suggestion
Fixed a series of broken URLS and typos in examples and documentation strings.
```

File: examples/saving_and_loading_data/coordinates_in_asdf.py [hunk 2]
```diff
@@ -4,7 +4,7 @@
 ========================================
 
 In this example we are going to look at saving and loading collections of
-coordinates with `asdf <https://asdf.readthedocs.io/en/latest/>`__.
+coordinates with `asdf <10.5281/zenodo.7448607>`__.
```
[the remark this answers is not in the document]
Comment (SuleimanFarah) [reply examples/saving_and_loading_data/coordinates_in_asdf.py:7]: Is preference given to the url in this case? The DOI is leads to a website that users can download the asdf library and is connected to a github repo which presents the documentation (here is the link: https://zenodo.org/record/6617557). I could revert the changes for this file if needed.

[the remark this answers is not in the document]
Comment (nabobalis) [reply examples/saving_and_loading_data/coordinates_in_asdf.py:7]: But this is not a URL that Simeone can click on. This is also meant to introduce people to the documentation for `asdf`. I don't see how a DOI helps here.

[the remark this answers is not in the document]
Comment (SuleimanFarah) [reply examples/saving_and_loading_data/coordinates_in_asdf.py:7]: Ok, I will amend this change in another commit.

**Review by nabobalis — APPROVED** [review]

**Comment by nabobalis** [comment]
Thanks @SuleimanFarah

**Pull request merged**